src/ai/ai_gui.cpp

```diff
@@ -640,15 +641,33 @@ struct ScriptTextfileWindow : public TextfileWindow {
 
 	ScriptTextfileWindow(TextfileType file_type, CompanyID slot) : TextfileWindow(file_type), slot(slot)
 	{
-		const char *textfile = GetConfig(slot)->GetTextfile(file_type, slot);
-		this->LoadTextfile(textfile, (slot == OWNER_DEITY) ? GAME_DIR : AI_DIR);
+		this->OnInvalidateData();
 	}
 
 	void SetStringParameters(int widget) const override
 	{
 		if (widget == WID_TF_CAPTION) {
 			SetDParam(0, (slot == OWNER_DEITY) ? STR_CONTENT_TYPE_GAME_SCRIPT : STR_CONTENT_TYPE_AI);
-			SetDParamStr(1, GetConfig(slot)->GetName());
+
+			if (GetConfig(slot)->GetInfo() != nullptr) {
+				SetDParamStr(1, GetConfig(slot)->GetInfo()->GetName());
+			} else {
+				char name[1024];
+				GetString(name, (slot == OWNER_DEITY) ? STR_AI_CONFIG_NONE : STR_AI_CONFIG_RANDOM_AI, lastof(name));
+				const char *script_name = stredup(name);
+				SetDParamStr(1, script_name);
```

Comment:
I'm pretty sure there's a memory leak here due to stredup, but is this branch of SetStringParameters even necessary any longer? Given the window will be removed if GetTextfile returns nullptr in OnInvalidateData

Reply:
I can't tell from just reading this bit of code whether it's possible the window may attempt a repaint with invalid script config/data, but at least it shouldn't be necessary to do the complex string formatting. Just `SetDParamStr(1, "");` and remove the entire GetString and stredup sequence. Nobody's going to see the window with missing text for more that one frame in the worst case.